keyboards/keychron/k12/rgb/ansi/keyboard.json

```diff
@@ -0,0 +1,161 @@
+            {"matrix": [1, 1], "x": 19, "y": 13, "flags": 4 },
+            {"matrix": [1, 2], "x": 31, "y": 13, "flags": 4 },
+            {"matrix": [1, 3], "x": 44, "y": 13, "flags": 4 },
+            {"matrix": [1, 4], "x": 56, "y": 13, "flags": 4 },
+            {"matrix": [1, 5], "x": 68, "y": 13, "flags": 4 },
+            {"matrix": [1, 6], "x": 81, "y": 13, "flags": 4 },
+            {"matrix": [1, 7], "x": 93, "y": 13, "flags": 4 },
+            {"matrix": [1, 8], "x": 106, "y": 13, "flags": 4 },
+            {"matrix": [1, 9], "x": 118, "y": 13, "flags": 4 },
+            {"matrix": [1, 10], "x": 131, "y": 13, "flags": 4 },
+            {"matrix": [1, 11], "x": 143, "y": 13, "flags": 4 },
+            {"matrix": [1, 12], "x": 155, "y": 13, "flags": 4 },
+            {"matrix": [1, 13], "x": 168, "y": 13, "flags": 4 },
+            {"matrix": [2, 0], "x": 5, "y": 26, "flags": 4 },
+            {"matrix": [2, 1], "x": 22, "y": 26, "flags": 4 },
+            {"matrix": [2, 2], "x": 34, "y": 26, "flags": 4 },
+            {"matrix": [2, 3], "x": 47, "y": 26, "flags": 4 },
+            {"matrix": [2, 4], "x": 59, "y": 26, "flags": 4 },
+            {"matrix": [2, 5], "x": 72, "y": 26, "flags": 4 },
+            {"matrix": [2, 6], "x": 84, "y": 26, "flags": 4 },
+            {"matrix": [2, 7], "x": 96, "y": 26, "flags": 4 },
+            {"matrix": [2, 8], "x": 109, "y": 26, "flags": 4 },
+            {"matrix": [2, 9], "x": 121, "y": 26, "flags": 4 },
+            {"matrix": [2, 10], "x": 134, "y": 26, "flags": 4 },
+            {"matrix": [2, 11], "x": 146, "y": 26, "flags": 4 },
+            {"matrix": [2, 13], "x": 166, "y": 26, "flags": 4 },
+            {"matrix": [3, 0], "x": 8, "y": 38, "flags": 4 },
+            {"matrix": [3, 2], "x": 28, "y": 38, "flags": 4 },
+            {"matrix": [3, 3], "x": 40, "y": 38, "flags": 4 },
+            {"matrix": [3, 4], "x": 53, "y": 38, "flags": 4 },
+            {"matrix": [3, 5], "x": 65, "y": 38, "flags": 4 },
+            {"matrix": [3, 6], "x": 78, "y": 38, "flags": 4 },
+            {"matrix": [3, 7], "x": 90, "y": 38, "flags": 4 },
+            {"matrix": [3, 8], "x": 103, "y": 38, "flags": 4 },
+            {"matrix": [3, 9], "x": 115, "y": 38, "flags": 4 },
+            {"matrix": [3, 10], "x": 128, "y": 38, "flags": 4 },
+            {"matrix": [3, 11], "x": 140, "y": 38, "flags": 4 },
+            {"matrix": [3, 13], "x": 165, "y": 38, "flags": 4 },
+            {"matrix": [4, 0], "x": 2, "y": 51, "flags": 4 },
+            {"matrix": [4, 1], "x": 17, "y": 51, "flags": 4 },
+            {"matrix": [4, 2], "x": 33, "y": 51, "flags": 4 },
+            {"matrix": [4, 6], "x": 79, "y": 51, "flags": 4 },
+            {"matrix": [4, 10], "x": 126, "y": 51, "flags": 4 },
+            {"matrix": [4, 11], "x": 141, "y": 51, "flags": 4 },
+            {"matrix": [4, 12], "x": 156, "y": 51, "flags": 4 },
+            {"matrix": [4, 13], "x": 171, "y": 51, "flags": 4 }
```

Comment:
those x,y coords are a bit sus ( not covering the full range). Have you verified them?

Reply:
Matches the old one, I went lazy: https://github.com/SonixQMK/qmk_firmware/blob/917a76c2b5c906f7f03ab07fc32399dc6ce9af15/keyboards/keychron/k12/rgb/ansi/ansi.c#L23 
Should I redo it from scratch?

Reply:
a quick plot with https://xelus.netlify.app/guides/kle_to_rgb_parser suggests they need to be redone
![Screenshot from 2024-08-10 14-16-58](https://github.com/user-attachments/assets/b9fc8d9b-2a8e-4a6a-9450-fea19193bbfe)

```
{  0,0  }, { 16,0  }, { 32,0  }, { 48,0  }, { 65,0  }, { 81,0  }, { 97,0  }, {113,0  }, {129,0  }, {145,0  }, {161,0  }, {178,0  }, {194,0  }, {218,0  }, 
{  4,16 }, { 24,16 }, { 40,16 }, { 57,16 }, { 73,16 }, { 89,16 }, {105,16 }, {121,16 }, {137,16 }, {153,16 }, {170,16 }, {186,16 }, {202,16 }, {218,16 }, 
{  6,32 }, { 28,32 }, { 44,32 }, { 61,32 }, { 77,32 }, { 93,32 }, {109,32 }, {125,32 }, {141,32 }, {157,32 }, {174,32 }, {190,32 }, {216,32 }, { 10,48 }, 
{ 36,48 }, { 52,48 }, { 69,48 }, { 85,48 }, {101,48 }, {117,48 }, {133,48 }, {149,48 }, {165,48 }, {182,48 }, {212,48 }, {  2,64 }, { 22,64 }, { 42,64 }, 
{103,64 }, {163,64 }, {184,64 }, {204,64 }, {224,64 }, 
```

good idea to also fix the flags while you're there https://github.com/SonixQMK/qmk_firmware/blob/sn32_develop/docs/features/rgb_matrix.md#flags-flags